Automattic/wp-calypso
# Components: Adds LoggedOutForm component

In an effort to factor the logged out link that shows at the bottom of the form at `/start/account/user` and to allow reuse of the logged out form styles, this PR refactors the `signup/logged-out-form` component and brings much of its functionality to `client/components/logged-out-form-container` component.

cc @mtias and @rickybanister for design review, @lezama, and @drewblaisdell for code review.

To test:
- Checkout `add/components-logged-out-form` branch
- Test invite flows
  - On a WP.com site, invite a test user by going to `$site/wp-admin/users.php?page=wpcom-invite-users`
  - In the invite email, make note of the invitation key
  - Go to `/accept-invite/$site/$invite_key`
  - While logged in or out, can you accept an invite? Are there errors? Does everything look OK?
- Test user creation
  - Go to `/start/account/user`
  - Attempt to create a user
  - Does everything work and look good?
- Test site creation
  - Go to `/start/site-user/site`
  - Does everything look ok? Are there any errors?

__Note:__ You'll notice that the `/log-in` form is off centered, and during testing you will likely run into a JS error if you try to submit the form on `/log-in`. This is not related to this PR and `/log-in` is only enabled in `development`.

__Also note:__ In some screenshots you'll notice there's quite a bit of padding between the form footer and the input directly above it. This seems to be caused by:

```
.validation-fieldset + .logged-out-form__footer {
    margin-top: 20px;
}
```

I intentionally added a screenshot with an invalid form field so that you could see what that would look like.

__After screenshots:__
![screen shot](https://cloud.githubusercontent.com/assets/1126811/11756987/5e61976c-a023-11e5-95b2-840e80d74a39.png)
![screen shot 1](https://cloud.githubusercontent.com/assets/1126811/11756988/5e626dae-a023-11e5-805d-9c2eb599edda.png)
![screen shot 2](https://cloud.githubusercontent.com/assets/1126811/11756990/5e632f96-a023-11e5-82fd-2ea04679dcac.png)
![screen shot 3](https://cloud.githubusercontent.com/assets/1126811/11756986/5e6000c8-a023-11e5-804b-57ecc35ced65.png)
![screen shot 4](https://cloud.githubusercontent.com/assets/1126811/11756989/5e628f00-a023-11e5-9af8-2406ac924788.png)
![screen shot](https://cloud.githubusercontent.com/assets/1126811/11758412/5129b3f6-a030-11e5-8c4b-48674df9faec.png)
